docs/message-queues.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[message-quques]]
+== Message Queues
+
+The Node.js Agent will automatically create spans for any activity to/from your Amazon SQS message queues.  In order to have those spans recorded, your message queue activity must  occur during a transaction.
+
+If you're performing queue operations during an HTTP request from a <<compatibility-frameworks,supported framework>>     the agent will start a transaction automatically.  However, if you're performing queue operations in a stand alone program (such as a message processor) you'll need to use the Node.js Agent's <<apm-start-transaction,`startTransaction()`>> method to create transactions for your messages.
+
+You can see an example of this in the following code sample.
+
+[source,js]
+----
+const apm = require('elastic-apm-node').start({/*...*/})
+const AWS = require('aws-sdk');
+// Set the region
+AWS.config.update({region: 'us-west'});
+
+// Create an SQS service object
+const sqs = new AWS.SQS({apiVersion: '2012-11-05'});
+
+/* ... */
+
+const transaction = apm.startTransaction("Process Messages", 'cli')
+sqs.receiveMessage(params, function(err, data) {
+  if(err) {
+    console.log("Receive Error", err);
+  } else {
+    console.log(`Length: ${data.Messages.length}`)
+    /* process messages */
+  }
+  // end the transaction
+  transaction.end()
+})
+----
+
+In the above code you can see we've started a transaction before calling the `sqs.receiveMessage` method, and ended it _after_ the queue's processing callback finishes executing.  The will ensure a transaction is active while processing your queue messages.
+
+[float]
+[[message-queues-distributed-tracing]]
+=== Distributed Tracing and Messaging Queues
+
+Similarly, if you want to connect your queue scheduling and queue processing via distributed tracing, you'll need to use the Node.js Agent's API to _store_ a `traceparent` header with your queue message and then provide that `traceparent` header when starting the new transaction.
```

Comment:
It might not be immediately clear what we're referring to when we say, "similarly". I would simplify to something like:

```suggestion
To enable queue scheduling and queue processing with distributed tracing, use the Node.js Agent's API to _store_ a `traceparent` header with your queue message; then, provide that `traceparent` header when starting a new transaction.
```